File: in_Java/1003 Check If Word Is Valid After Substitutions.java
class Solution {
    public boolean isValid(String S) {
        Stack<Character> st = new Stack<>();
        for(int i = 0; i < S.length(); i++){
            if(st.size() < 2){
                st.push(S.charAt(i));
                continue;
            }
            char pre = st.pop();
            char two = st.pop();
            if(S.charAt(i) == 'c' && pre == 'b' && two == 'a') continue;
            else{
                st.push(two);st.push(pre);st.push(S.charAt(i));
            }
        }
        return st.isEmpty();
    }
}

File: in_Python/1003 Check If Word Is Valid After Substitutions.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        st = []
        for x in s:
            if x == 'a' or x == 'b':
                st.append(x)
            else:
                if len(st) < 2: 
                    st.append(x)
                else:
                    if st[-1] == 'b' and st[-2] == 'a':
                        st.pop()
                        st.pop()
        return len(st) == 0
        